### PR TITLE
feat: add qwik cli e2e

### PR DESCRIPTION
**Pull request by dmitry-stepanenko**
This PR adds a separate `e2e-isolated` project with e2e tests that generate a new test qwik application each time and runs tests against it. More details are provided in the readme `e2e-isolated/qwik-e2e/README.md`

### CODE REVIEW

File: e2e/qwik-cli-e2e/tests/serve.spec.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    )
+  );
+  // wait for the arbitrary amount of time before the app is reloaded
+  await new Promise((r) => setTimeout(r, 2000));
+  expect((await getPageHtml(host)).querySelector('.container h1')?.textContent).toBe(
+    `So fantasticto have e2e tests here`
+  );
+}
+
+// TODO qwik-cli-e2e
```
Comment (shairez) [inline e2e/qwik-cli-e2e/tests/serve.spec.ts:66]: this could be removed right?

Comment (dmitry-stepanenko) [reply e2e/qwik-cli-e2e/tests/serve.spec.ts]: yep, leftover, thanks

**Comment by shairez** [comment]
Looks good @dmitry-stepanenko 

2 last things we need to modify are - 

1. Adding this to the CI testing process

2. Because you added "e2e" to pnpm workspace, we need to make sure it's excluded in the `.changeset/config` file, so it won't be scanned for changes etc

After that we could merge it, thanks!

(let me know if you need any help with either 1 or 2)

**Review by shairez — APPROVED** [review]
Thanks @dmitry-stepanenko 🙏

**Pull request merged**